src/sync/impl/sync_file.cpp

```diff
@@ -420,6 +426,15 @@ std::string SyncFileManager::legacy_realm_file_path(const std::string& local_use
     return path;
 }
 
+std::string SyncFileManager::legacy_local_identity_path(const std::string& local_user_identity, const std::string& realm_file_name) const
+{
+    auto escaped_file_name = validate_and_clean_path(realm_file_name);
+    std::string path_name = util::file_path_by_appending_component(user_directory(local_user_identity), escaped_file_name);
```

Comment:
Notice that `user_directory` calls `try_make_dir`, it will create the directory for the local identity path.